tests/foreman/cli/test_discoveryrule.py

```diff
@@ -0,0 +1,422 @@
+        max count option
+
+        @id: 590ca353-d3d7-4700-be34-13de00f46276
+
+        @Assert: Rule should be successfully created and has max_count set
+        as per given value
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @tier1
+    def test_positive_create_with_priority(self):
+        """Create Discovery Rule providing any number from range 1..100 for
+        priority option
+
+        @id: 8ef58279-0ad3-41a4-b8dd-65594afdb655
+
+        @Assert: Rule should be successfully created and has expected priority
+        value
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @tier1
+    def test_positive_create_disabled_rule(self):
+        """Create Discovery Rule in disabled state
+
+        @id: 8837a0c6-e19a-4c33-8b87-07b6f69dbb0f
+
+        @Assert: Disabled rule should be successfully created
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @tier1
+    def test_negative_create_with_invalid_name(self):
+        """Create Discovery Rule with invalid names
+
+        @id: a0350dc9-8f5b-4673-be88-a5e35d1f8ca7
+
+        @Assert: Error should be raised and rule should not be created
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @tier1
+    def test_negative_create_with_invalid_hostname(self):
+        """Create Discovery Rule with invalid hostname
+
+        @id: 0ae51085-30d0-44f9-9e49-abe928a8a4b7
+
+        @Assert: Error should be raised and rule should not be created
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @tier1
+    def test_negative_create_with_too_long_limit(self):
+        """Create Discovery Rule with too long host limit value
+
+        @id: 12dbb023-c963-4ead-a81e-ad53033de947
+
+        @Assert: Validation error should be raised and rule should not be
+        created
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @tier1
+    def test_negative_create_with_same_name(self):
+        """Create Discovery Rule with name that already exists
+
+        @id: 5a914e76-de01-406d-9860-0e4e1521b074
+
+        @Assert: Error should be raised and rule should not be created
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @tier1
+    def test_positive_delete(self):
+        """Delete existing Discovery Rule
+
+        @id: c9b88a94-13c4-496f-a5c1-c088187250dc
+
+        @Assert: Rule should be successfully deleted
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @tier1
+    def test_positive_update_name(self):
+        """Update discovery rule name
+
+        @id: 1045e2c4-e1f7-42c9-95f7-488fc79bf70b
+
+        @Assert: Rule name is updated
+
+        @caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_update_by_associating_org_loc(self):
```

Comment:
Yeah.. the test pointed above is for updating org_location only. So will update with name:
`test_positive_update_org_loc`